pkg/cloudevents/transport/httpb/transport.go

```diff
@@ -0,0 +1,238 @@
+package httpb
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/bindings/http"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/plugin/ochttp/propagation/tracecontext"
+	"net"
+	nethttp "net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	"github.com/cloudevents/sdk-go/pkg/event"
+)
+
+// Transport adheres to transport.Transport.
+var _ transport.Transport = (*Transport)(nil)
+
+const (
+	// DefaultShutdownTimeout defines the default timeout given to the http.Server when calling Shutdown.
+	DefaultShutdownTimeout = time.Minute * 1
+)
+
+// Transport acts as both a http client and a http handler.
+type Transport struct {
+	binding.BindingTransport
+	consumer transport.Receiver
+
+	// The encoding used to select the codec for outbound events.
+	Encoding Encoding
+
+	// ShutdownTimeout defines the timeout given to the http.Server when calling Shutdown.
+	// If nil, DefaultShutdownTimeout is used.
+	ShutdownTimeout *time.Duration
+
+	// Port is the port to bind the receiver to. Defaults to 8080.
+	Port *int
+	// Path is the path to bind the receiver to. Defaults to "/".
+	Path string
+
+	// Receive Mutex
+	reMu sync.Mutex
+	// Handler is the handler the http Server will use. Use this to reuse the
+	// http server. If nil, the Transport will create a one.
+	Handler           *nethttp.ServeMux
+	listener          net.Listener
+	server            *nethttp.Server
+	handlerRegistered bool
+	middleware        []Middleware
+	Target            *url.URL
+}
+
+func New(opts ...Option) (*Transport, error) {
+
+	t := &Transport{}
+	if err := t.applyOptions(opts...); err != nil {
+		return nil, err
+	}
+
+	if t.Requester == nil {
+		client := nethttp.DefaultClient
+		t.Requester = http.NewRequester(client, t.Target) // TODO: requester does not support dynamic update of target.
+	}
+
+	return t, nil
+}
+
+func (t *Transport) applyOptions(opts ...Option) error {
+	for _, fn := range opts {
+		if err := fn(t); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Send implements Transport.Send
+func (t *Transport) Send(ctx context.Context, e event.Event) (context.Context, *event.Event, error) {
+	switch t.Encoding {
+	case Default, Binary:
```

Comment:
What if, for default, you don't decorate at all the context and let the binding try the best encoding?

Reply:
the transport has been set to a mode as a supported feature in cloudevents.Transport. This plumbs that feature through